### PR TITLE
Avoid continuous setting of watchers by using a Persistent watch.

### LINKED ISSUE #44: Long running Idempotent task keeps creating watches and leads to Out Of Memory issues
SimpleQueue lists queued task and sets watcher to watch for new events in case of empty but if task is present and its picked up by some worker its not deleted in case of idempotent and all other workers goes into loop of watching again and again. If this single task is long processing other workers or threads if multithreaded keeps creating watcher instances.

### PR DESCRIPTION
**Pull request by Randgalt**
Avoid continuous setting of watchers by using a Persistent watch. This should fix the reported memory issue with many watchers being set.

Note: this requires ZooKeeper 3.6.x or above

Fixes #44

### CODE REVIEW

File: src/main/java/com/nirmata/workflow/details/WorkflowListenerManagerImpl.java [hunk 1]
```diff
@@ -21,55 +21,48 @@
...
 import java.io.IOException;
 
-public class WorkflowListenerManagerImpl implements WorkflowListenerManager
-{
+public class WorkflowListenerManagerImpl implements WorkflowListenerManager {
     private final PathChildrenCache completedTasksCache;
     private final PathChildrenCache startedTasksCache;
     private final PathChildrenCache runsCache;
-    private final ListenerContainer<WorkflowListener> listenerContainer = new ListenerContainer<>();
+    private final UnaryListenerManager<WorkflowListener> listenerContainer = StandardListenerManager.standard();
```
Comment (Randgalt) [inline src/main/java/com/nirmata/workflow/details/WorkflowListenerManagerImpl.java:36]: Note: changes here due to API changes in Curator. This is not directly related to the issue being fixed in this PR.

**Comment by gs-psharma** [comment]
any update on getting this released?

**Comment by Randgalt** [comment]
@JimBugwadia and @dtoledo67 are you OK with this?

**Comment by JimBugwadia** [comment]
@Randgalt - yes thanks! 

Also, are there any other dependencies, etc. we can update before a release?

**Comment by Randgalt** [comment]
Github updated Jackson for us so we're OK
